### DISCUSSION TITLE
can you proivde a demo about setting up dynamic meta data when sharing in LINE or FB?

### DISCUSSION THREAD

**Discussion by withgogo** [nuxt/nuxt#28669] [Questions]
I've struggled about this for couple of days. 

The question is:
1. I get the data from api requests, and assign to meta data, like title, keywords, description, og:title, og:image, by useSeoMeta/useHead
2. However, in below tools, it could get nothing about dynamic meta:
LINE tool: https://poker.line.naver.jp/
FB tool: https://developers.facebook.com/tools/debug/

So would anyone provide an example about this scenario?  
Please help!

**Comment by withgogo** [comment]
anyone can help?

**Reply by TheAlexLichter** [reply]
With the information given, not really. Either you provide a minimal reproduction (and code) for people to look at, or you might bring someone in the project on board to help you out debugging it.

**Reply by withgogo** [reply]
Here is the code & minimal reproduction.
https://github.com/withgogo/ec-ssr
https://stackblitz.com/~/github.com/withgogo/ec-ssr

**Reply by TheAlexLichter** [reply]
@withgogo this is a whole application and not minimal I'm afraid. Please remove all code that is unrelated except setting the dynamic meta data. Maybe start with https://nuxt.new/

**Reply by withgogo** [reply]
@manniL Hi, I'm sorry for the inconvenience. Please check the code below, thanks:
https://codesandbox.io/p/devbox/jolly-zeh-ckkmkn

**Reply by TheAlexLichter** [reply]
Thanks! Yes, there are some usage errors. Let me share some untested example code.

Issues you had were mostly that you didn't use the result from useAsyncData/useFetch and didn't work with them ("in a reactive way").

```vue

<script setup>
const route = useRoute()
const params = computed(() => ({
    storeId: route.query.storeId,
    id: route.query.id,
}))

const { data } = useFetch('http://ec-app.pqhealthy.com/api/h5/goods/items', {
    params,
    transform: (res) => {
        return res.obj
    },
})

useSeoMeta({
    title: () => data?.goodsName,
    ogTitle: () => data?.goodsName,
    description: () => data?.seoDescription,
    ogDescription: () => data?.seoDescription,
    ogImage: () => data?.goodsMainPhoto?.fullPath,
    twitterCard: 'summary_large_image',
})
</script>

<template>
    <h1>Goods</h1>
    <div v-if="data.goodsInfo">
        <div>title: {{ data.goodsInfo.goodsName }}</div>
        <div>description:{{ data.goodsInfo.seoDescription }}</div>
        <img
            :src="data.goodsInfo?.goodsMainPhoto?.fullPath"
            width="100"
            height="100"
        />
    </div>
    <NuxtLink to="/goods?id=2331">Other param</NuxtLink>
    <NuxtLink to="/">Back</NuxtLink>
</template>
```

**Reply by withgogo** [reply]
@manniL Thanks for you reply. However, it didn't work. there is no dynamic meta tag like "og:title,og:decription, og:image" at all.
<img width="1505" alt="image" src="https://github.com/user-attachments/assets/dcfd4b5b-a453-4ac7-a6e8-46d91310a847">
Please check the newest code  by your suggestion:https://codesandbox.io/p/devbox/jolly-zeh-ckkmkn

**Reply by TheAlexLichter** [reply]
`await` useFetch.

**Comment by withgogo** [comment]
If I use the useServerSeoMeta, it won't displayed in the <head> section in the page. But can be crawled by Robots?

**Reply by TheAlexLichter** [reply]
`useServerSeoMeta` is only useful it the info will never change on the client-side.
